backend/module/eCampApi/src/eCampApi/V1/Factory/Config.php

```diff
@@ -0,0 +1,264 @@
+<?php
+
+namespace eCampApi\V1\Factory;
+
+class Config {
+    private $nameSingular;
+    private $namePlural;
+    private $routeName;
+    private $routeIdentifierName;
+    private $endpoint;
+    private $entityHttpMethods;
+    private $collectionHttpMethods;
+    private $collectionQueryWhitelist = ['page_size', 'page'];
+    private $inputFilterItems = [];
+
+    public function __construct(string $nameSingular) {
+        $this->nameSingular = $nameSingular;
+    }
+
+    public static function Create(string $nameSingular, ?string $namePlural = null): Config {
+        $factory = new self($nameSingular);
+        if (isset($namePlural)) {
+            $factory->setNamePlural($namePlural);
+        }
+
+        return $factory;
+    }
+
+    public function getNameSingular(): string {
+        return $this->nameSingular;
+    }
+
+    public function setNameSingular(string $nameSingular): Config {
+        $this->nameSingular = $nameSingular;
+
+        return $this;
+    }
+
+    public function getNamePlural(): string {
+        return $this->namePlural ?? ($this->nameSingular.'s');
+    }
+
+    public function setNamePlural(string $namePlural): Config {
+        $this->namePlural = $namePlural;
+
+        return $this;
+    }
+
+    public function getRouteName(): string {
+        return $this->routeName ?? ('e-camp-api.rest.doctrine.'.self::toKebabCase($this->nameSingular));
+    }
+
+    public function setRouteName(string $routeName): Config {
+        $this->routeName = $routeName;
+
+        return $this;
+    }
+
+    public function getRouteIdentifierName(): string {
+        return $this->routeIdentifierName ?? (lcfirst($this->nameSingular).'Id');
+    }
+
+    public function setRouteIdentifierName(string $routeIdentifierName): Config {
+        $this->routeIdentifierName = $routeIdentifierName;
+
+        return $this;
+    }
+
+    public function getEndpoint(): string {
+        return $this->endpoint ?? self::toKebabCase($this->getNamePlural());
```

Comment:
I would put the logic into the build method, not in the getter

Reply:
#1018